quinn/src/streams.rs

```diff
@@ -149,6 +151,25 @@ where
         conn.wake();
     }
 
+    /// Monitor peer's STOP_SENDING issue for this stream
+    ///
+    /// This will return the reason of reset when receiving
+    pub fn poll_stopped(&mut self, cx: &mut Context) -> Poll<Option<VarInt>> {
```

Comment:
Why does this return an `Option`? What does it mean if `None` is returned?

I'd also prefer to expose a future, rather than a `poll_` function.

Reply:
IIRC in the end this channels into hyper API where it needs to be part of a named future, which I think means we can't only have an async API.

Reply:
We could provide a named future, or both a future and a (`#[doc(hidden)]`?) `poll_` fn as we currently do for `poll_finish`.

Reply:
I was concerned about the future non-resolvabity after some events (like closing, errors). But after all this should be handled by the user through other API. And yes, `None` should have been replaced by  an `Result::Err()`.

But I got rid of it.

Also, I implemented a `Stopped` future similar to `Finish`